### PR TITLE
Preference UX improvements

### PR DESCRIPTION
**Pull request by maniac103**
This started off as an improvement for the DE translation, but while working on the translation I found a number of improvement areas in the preference handling code, which I also fixed. I thought about adding them in a separate PR, but unfortunately they touch strings as well, so would conflict with the DE translation update.
If you prefer, I can split this PR to merge the translation updates first, then submit the preference code updates in a separate PR later.

### CODE REVIEW

File: mobile/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -15,76 +15,79 @@
...
-    <string name="settings_openhab_demomode_summary">Betreibt openHAB im Demo Modus</string>
-    <string name="settings_openhab_theme">openHAB Thema</string>
-    <string name="settings_openhab_appversion">openHAB App Version</string>
-    <string name="settings_openhab_sslcert">Ignoriere SSL Zertifikat</string>
-    <string name="settings_openhab_sslcert_summary">Ignoriere Zertifikatsvalidierung beim SSL-Handshake</string>
-    <string name="settings_openhab_sslhost">Ignoriere SSL Hostnamen</string>
-    <string name="settings_openhab_sslhost_summary">Ignoriere Hostname-Prüfung beim SSL-Handshake</string>
+    <string name="settings_openhab_password_summary">openHAB-Passwort</string>
+    <string name="settings_openhab_sslclientcert">SSL-Client-Zertifikat</string>
+    <string name="settings_openhab_sslclientcert_howto">Erläuterungen zum SSL-Client-Zertifikat</string>
```
Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: Maybe better: "Erklärungen zur SSL-Client-Authentifizierung"

File: mobile/src/main/res/values-de/strings.xml [hunk 2]
```diff
@@ -15,76 +15,79 @@
...
+    <string name="settings_openhab_screentimeroff">Ruhezustand deaktivieren</string>
+    <string name="settings_openhab_screentimeroff_summary">Schaltet das automatische Aktivieren des Ruhezustands nach Inaktivität ab, wenn die openHAB-App läuft</string>
+    <string name="settings_openhab_demomode">Demo-Modus</string>
+    <string name="settings_openhab_demomode_summary">Demo-Daten an Stelle von realen Server-Daten verwenden</string>
+    <string name="settings_openhab_theme">App-Design</string>
```
Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: Ich würde hier bei "Thema" bleiben oder "Design" verwenden.

File: mobile/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (maniac103) [inline mobile/src/main/res/values-de/strings.xml]: Das wird bestimmt etwas zu lang für kleine Displays. Eventuell 'HowTo für SSL-Client-Zertifikat' + Summary ('Für Erläuterungen zur Einrichtung von SSL-Client-Zertifikaten hier berühren')?

File: mobile/src/main/res/values-de/strings.xml (hunk 2 above)
Comment (maniac103) [inline mobile/src/main/res/values-de/strings.xml]: 'Thema' assoziiere ich immer mit dem, was im Englischen 'topic' heißt. 'Design' (ohne 'App-' davor) klingt aber OK.

File: mobile/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: ...hier tippen?

Comment (maniac103) [inline mobile/src/main/res/values-de/strings.xml]: :+1:

File: mobile/src/main/res/values-de/strings.xml [hunk 3]
```diff
@@ -15,78 +15,77 @@
...
+    <string name="settings_openhab_appversion">openHAB-App-Version</string>
+    <string name="settings_openhab_sslcert">SSL-Zertifikat ignorieren</string>
+    <string name="settings_openhab_sslcert_summary">Beim SSL-Handshake alle Zertifikate unabhänging von ihrer Gültigkeit akzeptieren</string>
```
Comment (maniac103) [inline mobile/src/main/res/values-de/strings.xml]: Note to self: 'unabhängig'.

File: mobile/src/main/res/values-de/strings.xml (hunk 2 above)
Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: Ich bevorzuge "Thema". Aus wikipedia:

> Thema (griechisch: θέμα, théma ‚Behauptung, Ausspruch‘; teils auch Thematik, Pl. Themen) bezeichnet:
[...]
    ein Paket von Einstellungen und Bildern, die Funktion und Aussehen der grafischen Benutzeroberfläche (GUI) von Computerprogrammen festlegen, siehe Skin (Computer)

Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: Let's get some more opinions: https://forum.f-droid.org/t/how-to-translate-theme/853

**Comment by digitaldan** [comment]
Hi, thanks for the PR!  @mueller-ma since German is not my native tongue, can you give the 👍  on the translation?  Thanks!

**Comment by mueller-ma** [comment]
@digitaldan Not yet. I will aprove this PR, when the suggestions are commited

File: mobile/src/main/res/values-de/strings.xml (hunk 2 above)
Comment (maniac103) [inline mobile/src/main/res/values-de/strings.xml]: Hast du den Wikipedia-Link mal weiterverfolgt? ;-) Aus https://de.wikipedia.org/wiki/Skin_(Computer):
`Ein Skin (engl. ‚Haut‘, ‚Verkleidung‘), manchmal auch Design oder Feathers (‚Federn‘), ist [...]`

Der Duden kennt auch nur 'topic': http://www.duden.de/rechtschreibung/Thema

Wenn 'Design' (das hatte ich aus der Übersetzung der LineageOS-Theme-Engine übernommen) nicht ok ist (aber warum nicht?), würde ich persönlich eher 'Theme' stehen lassen ... dict.cc neigt auch dazu: http://www.dict.cc/?s=theme (vorletztes Substantiv).

Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: Auf fdroid ist einer für Thema, ein andrer für Aussehen. Bleiben wir bei Design

**Comment by digitaldan** [comment]
@maniac103 @mueller-ma  how is this PR coming?

**Comment by maniac103** [comment]
I'm currently on vacation, I'll update the PR as soon as I'm back (start of next week).

**Comment by maniac103** [comment]
Sorry for the delay, was a bit busy :-(
I've updated the PR for the requested changes and rebased it onto master.

File: mobile/src/main/res/values-de/strings.xml [hunk 4]
```diff
@@ -15,78 +15,78 @@
     <string name="app_discovery">Starte Suche</string>
     <!-- Main menu items -->
     <string name="mainmenu_openhab_preferences">Einstellungen</string>
-    <string name="mainmenu_openhab_selectsitemap">Wähle Sitemap</string>
-    <string name="mainmenu_openhab_writetag">Schreibe NFC Tag</string>
-    <string name="mainmenu_openhab_info">openHAB Info</string>
-    <string name="mainmenu_openhab_clearcache">Lösche Bildercache</string>
+    <string name="mainmenu_openhab_selectsitemap">Sitemap-Auswahl</string>
```
Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: Please include a translation of "default"

Comment (maniac103) [inline mobile/src/main/res/values-de/strings.xml]: This wasn't present before either, though (see old line 18). Maybe 'Standard-Sitemap auswählen'?

Comment (mueller-ma) [inline mobile/src/main/res/values-de/strings.xml]: I saw that is was wrong, but didnt want to make an extra pr, when you are working on this. 'Standard-Sitemap auswählen' sounds good

**Comment by mueller-ma** [comment]
Can you replace OpenHABPreferenceActivity Line 203 and 208 the stars with "●" (U+25CF)? It's centered vertically, the asterisk not.

**Comment by mueller-ma** [comment]
I have seen that the theme selector doesnt use the theme colors.
And android studio recommends:
```
-            android:title="@string/settings_ringtone"></RingtonePreference>
+            android:title="@string/settings_ringtone" />
```

**Comment by maniac103** [comment]
> I have seen that popups dont use theme colors, not introduced by your commit, but maybe you can fix it.

Popups as in AlertDialogs? Or ListPreference dialogs? I think we have to switch to the support library dialogs to fix it.

**Comment by mueller-ma** [comment]
Theme selection, user name and password config have teal buttons

**Comment by maniac103** [comment]
OK, there's about 99% probability a switch to the support preference lib will fix this. Will look into it.

**Comment by maniac103** [comment]
Oh, I just notice it works fine for me, that is, those prefs use the orange accent color for me. What Android version are you using?

**Comment by mueller-ma** [comment]
Your master branch on Android 5.1.1:
user name: Orange text line, teal buttons

Stable from Play on Android 7.1.2
user name: Orange text line, orange buttons

**Comment by maniac103** [comment]
I'm using my build on 7.1.2 at the moment, so it's probably the older API. I'll check it in emulator.

**Comment by maniac103** [comment]
Fixed the asterisks.

I also have a patch to use the support prefs (see https://github.com/maniac103/openhab.android/tree/support-prefs), which fixes the accent color, but EditTextPreference doesn't simply accept the android:inputType anymore. I didn't find an elegant solution to that, so I left it out for now.

**Comment by mueller-ma** [comment]
Cannot find "password_placeholder" in the preferences.

**Comment by maniac103** [comment]
Huh? It's here: https://github.com/openhab/openhab.android/pull/337/files#diff-2bd3394bbec37bbcc0aafdd9a3cc0fa5R94

**Comment by mueller-ma** [comment]
And where do you set it?

**Comment by maniac103** [comment]
I guess I don't understand ... what do you mean by 'set'? password_placeholder is only used as summary text.

**Comment by mueller-ma** [comment]
You define it here https://github.com/openhab/openhab.android/pull/337/files#diff-2bd3394bbec37bbcc0aafdd9a3cc0fa5R94 but where do you set it as summary?

**Comment by maniac103** [comment]
Here: https://github.com/openhab/openhab.android/pull/337/files#diff-b8963974f554d6698e36b7b2a0fa790aR178

**Comment by mueller-ma** [comment]
Sry, I searched in Github and the diff didnt load

**Comment by digitaldan** [comment]
is this ready to be merged?

**Comment by mueller-ma** [comment]
I think so

**Comment by maniac103** [comment]
From my side it's ready.

**Comment by digitaldan** [comment]
So one thing I have noticed when testing, and I'm not sure if this is due to this change (I think it is), but if I have a host that can not be connected to in the local field, and the remote field is blank, then auto discovery is still attempted.  I would assume when a local value is entered, we disable auto discovery.

**Comment by mueller-ma** [comment]
From the play store version:

> 09-29 20:19:04.990 21637-21637/org.openhab.habdroid D/OpenHABTracker: Checking reachability of http://openhab/
> 09-29 20:19:05.138 21637-21689/org.openhab.habdroid E/OpenHABTracker: Host is unresolved: openhab
> 09-29 20:19:05.142 21637-21637/org.openhab.habdroid E/Util: normalizeUrl: invalid URL
> 09-29 20:19:05.192 21637-21693/org.openhab.habdroid D/OpenGLRenderer: Use EGL_SWAP_BEHAVIOR_PRESERVED: true
> 09-29 20:19:05.200 21637-21637/org.openhab.habdroid D/Atlas: Validating map...
> 09-29 20:19:05.220 21637-21696/org.openhab.habdroid I/AsyncServiceResolver: Discovering service _openhab-server-ssl._tcp.local.
> 09-29 20:19:05.231 21637-21696/org.openhab.habdroid I/AsyncServiceResolver: IP: fe80::some-ip%wlan0
> 09-29 20:19:05.231 21637-21696/org.openhab.habdroid I/AsyncServiceResolver: Is IPV4 = false
> 09-29 20:19:05.231 21637-21696/org.openhab.habdroid I/AsyncServiceResolver: IP: some-ip
> 09-29 20:19:05.231 21637-21696/org.openhab.habdroid I/AsyncServiceResolver: Is IPV4 = true
> 09-29 20:19:05.231 21637-21696/org.openhab.habdroid I/AsyncServiceResolver: Selected some-ip

**Comment by digitaldan** [comment]
Hi, I'm good with merging, @maniac103 can you a) fix the conflict that has occured due to other PRs and 2) Squash and sign your commit according to https://github.com/openhab/openhab.android/blob/master/CONTRIBUTING.md#sign-your-work ?

Thanks!

**Comment by maniac103** [comment]
@digitaldan Adding signed-off-by and fixing the conflict is no problem (will do so tomorrow), but do you really want me to squash all those commits? I tried really hard to split them up in logical units...

**Comment by digitaldan** [comment]
> but do you really want me to squash all those commits? I tried really hard to split them up in logical units.

They get squashed when we merge in any case.  Next time it's best to have separate PRs instead.  I would just concat all the messages into a single commit message that is signed.

**Comment by digitaldan** [comment]
or you could break this into 2 PR's now as well.

**Comment by maniac103** [comment]
> They get squashed when we merge in any case

**Comment by digitaldan** [comment]
>Please excuse my insistence, but: why? As you've stated you're OK with the code changes, why destroy the code history here?

We always squash commits within a single PR in OH projects, otherwise our commit history becomes difficult to manage.  We really only care about the changes within a PR, not the many commits that make up a PR.   The volume of commits plagued OH1 and we started squashing once OH2 was started.  We used to require users to squash once the PR was approved, but recently Github has added this feature in as the default merge action as this was a very common practice. So normally I don't have to ask, but since you need to sign your commit(s), this would seem natural for you to do so you only need to modify one.  If you want to modify all commit messages that would work too.

 >So you want me to submit the translation stuff (first 3 commits), wait for you to merge it and submit another PR with the remaining commits?

They should be 2 as they don't have anything in common, but I also don't want to hold this up and make more work as I was very late reviewing this and its more important it gets merged.

**Comment by maniac103** [comment]
OK, here's the first 2 PRs:
- https://github.com/openhab/openhab.android/pull/397
- https://github.com/openhab/openhab.android/pull/398

Will submit the DE translation once those are in, and update this PR for the preference stuff after that one is in.

**Pull request merged**